### ISSUE TITLE
Is it possible to remove virtualbox dependency?

### ISSUE THREAD

**Issue by rahuldeepattri** [myspaghetti/macos-virtualbox#240]
Many thanks for creating this awesome script! But sadly, I cannot install virtualbox for various reasons, most sad reason being Oracle can move away from open source licensing at anytime. Is there any way I could use Vmware workstation as an alternative and is it possible to remove virtualbox dependency?

**Comment by myspaghetti** [comment]
So you don't want to use the current software because worries about _non-open-source_ licensing, and you want to move to _non-open-source_ software.

Outside of that flawed reasoning, to answer your question, the script is almost trivial to adapt to QEMU (GPLv2) and VMWare vSphere (proprietary license) because they have a similar mechanism for configuring virtual machines from the command line and for sending scancodes to the virtual machine.

However, vSphere is proprietary and QEMU does not provide a GUI with a binary release on Windows or macOS.

Still, if someone forks this project for Linux-only, I heartily recommend QEMU because it solves the audio and EDID issues, and with KVM it solves performance issues.

**Comment by rahuldeepattri** [comment]
Thanks for your clarification @myspaghetti. I understand your point and I'll need to do my research first before asking a feature here.

Honestly speaking, I am a novice with VMs. My understanding as of now is that using VM Workstation 14 player I could run a virtual machine with Ubuntu OS on it. It needs a *.vdmk file, which I had created from disc image. Last time I checked VMware Workstation was a freeware for non-commercial use.

My purpose to install macOS and run it VM Workstation is to try out Swift and try out making native iOS apps. I am not using it for any commercial use. I cannot use the VirtualBox because I have restrictions for it and do not want to install on my machine.

I hoped that using your script I would have been able to create VM with macOS as the Operating System & then I could run that VM on VM workstation player.

I am not sure how vSphere comes into the picture or what underlying technology changes are needed to create *.vdmk file that can be used to create a VM & run it on VM workstation player.

Hope my explanation make much clearer what I originally intended to ask.
